modules/@apostrophecms/login/index.js

```diff
@@ -353,8 +353,11 @@ module.exports = {
             throw self.apos.error('notfound');
           }
 
+          const allowedFields = new Set([ 'role' ]);
```

Comment:
I apologize for the confusion, which started on our end because we didn't discuss the security implications enough internally first.

We don't need this `allowedFields` mechanism, and we don't need `req.query.additionalFields` either, because the only fields that should ever be returned are those in the `minimumWhoamiFields` and `whoamiFields` options. And those should always be returned.

We appreciate the contribution! The good news is you can simplify it now and remove the tests that relate to `req.query.additionalFields` because the feature is not needed.